netcore/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.Mono.cs

```diff
@@ -112,11 +112,17 @@ internal static bool IsBitwiseEquatable<T> ()
 		}
 
 		[Intrinsic]
-		internal static unsafe bool ObjectHasComponentSize (object obj)
+		internal static bool ObjectHasComponentSize (object obj)
 		{
 			throw new NotImplementedException ();
 		}
 
+		[Intrinsic]
+		internal static bool ElementTypeHasReferences (object obj)
+		{
```

Comment:
This could be ObjectHasReferences.

Reply:
It checks `has_reference` for `element_class`. `has_reference` for Array itself always returns the same value despite the inner element type.

Reply:
Strange, its initialized in mono_class_create_bounded_array () as:
	klass->has_references = MONO_TYPE_IS_REFERENCE (m_class_get_byval_arg (eclass)) || m_class_has_references (eclass)? TRUE: FALSE;

Reply:
Also, the default implementation below looks wrong in that case.